holidays/countries/nepal.py

```diff
@@ -0,0 +1,304 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.groups import (
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class Nepal(
+    HolidayBase,
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+):
+    """Nepal holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Nepal>
+        * <https://narayanilawfirm.org.np/list-of-public-holidays-in-nepal-2079/>
+        * <https://www.timeanddate.com/holidays/nepal/>
+    """
```

Comment:
Include the source for 2019-2020 holiday cuts here as well:
```suggestion
    References:
        * <https://en.wikipedia.org/wiki/Public_holidays_in_Nepal>
        * <https://narayanilawfirm.org.np/list-of-public-holidays-in-nepal-2079/>
        * <https://www.timeanddate.com/holidays/nepal/>
        * <https://kathmandupost.com/national/2018/03/29/government-trims-public-holidays-by-22-days>
        * <https://kathmandupost.com/national/2021/02/13/two-years-after-shortening-public-holiday-list-government-starts-adding-to-it-again>
    """
```